File: Backend/src/main/java/com/Trabajo_Final_Beltran/dto/response/PageResponse.java
package com.Trabajo_Final_Beltran.dto.response;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PageResponse<T> {

  private List<T> content;

  private int pagina;

  private int size;

  private long totalElementos;

  private int totalPaginas;

  private boolean primera;

  private boolean ultima;
}

File: frontEnd/src/app/core/models/page-response.ts
/**
 * Respuesta paginada del backend (PageResponse<T>).
 * Campos exactos devueltos por el backend.
 */
export interface PageResponse<T> {
    content: T[];
    pagina: number;
    size: number;
    totalElementos: number;
    totalPaginas: number;
    primera: boolean;
    ultima: boolean;
}
